### PR TITLE
Add BroadcastChannel API

### PR DESCRIPTION
**Pull request by falsandtru**
Fixes https://github.com/Microsoft/TypeScript/issues/15407

I cannot add the following code. How can I do that?

```ts
declare var BroadcastChannel: {
    prototype: BroadcastChannel;
    new(name: string): BroadcastChannel;
}
```

### CODE REVIEW

**Comment by falsandtru** [comment]
@zhengbli Is there a way to do that?

**Comment by mhegazy** [comment]
@falsandtru can you update the baselines.

File: inputfiles/addedTypes.json [hunk 1]
```diff
@@ -1,5 +1,60 @@
...
+        "interface": "BroadcastChannel",
+        "signatures": ["new(name: string): BroadcastChannel"]
+    },
+    {
+        "kind": "interface",
+        "name": "BroadcastChannel",
+        "extends": "EventTarget",
+        "properties": [
+            {
+                "name": "name",
```
Comment (mhegazy) [inline inputfiles/addedTypes.json]: `        "readonly": true,`

Comment (falsandtru) [inline inputfiles/addedTypes.json]: done.

**Comment by mhegazy** [comment]
CI still not happy :)

**Comment by falsandtru** [comment]
@mhegazy As I said at https://github.com/Microsoft/TSJS-lib-generator/pull/238#issue-224733274, seems like we cannot add constructors. Any ideas?

**Comment by falsandtru** [comment]
I just found the way. Please review it.

**Comment by falsandtru** [comment]
@mhegazy you should merge prs before getting conflicts.

**Comment by Andries-Smit** [comment]
Can we make it use a generic?

With the default generic it could `any` or a specific structure.
``` ts
interface BroadcastChannelMessageEvent<Message> extends MessageEvent {
    data: Message;
}
interface BroadcastChannel<Message=any> extends EventTarget {
    onmessage: (message: BroadcastChannelMessageEvent<Message>) => void;
}

**Pull request merged**